src/app/shared/loading.interceptor.ts

```diff
@@ -0,0 +1,24 @@
+import { HttpEvent, HttpHandler, HttpInterceptor, HttpRequest, HttpResponse } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { delay, tap } from 'rxjs/operators';
+import { LoaderService } from '@shared/loader.service';
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class LoadingInterceptor implements HttpInterceptor {
+
+  constructor(public loaderService: LoaderService) {
```

Comment:
Why `public`?

Reply:
It's a leftover, thank you for noticing. I tried to understand why it didn't work.